### PR TITLE
[Bullet] Inertial Pose Update

### PR DESCRIPTION
**Pull request by Lobotuerk**
# 🎉 New feature

Should resolve https://github.com/ignitionrobotics/ign-physics/pull/208#discussion_r592823935

## Summary
Updated Bullet's pose calculation to also use the `Pose` element of the `<inertial>` sdf tag.

## Test it
No idea, open for suggestions

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: bullet/src/SDFFeatures.cc [hunk 1]
```diff
@@ -147,8 +149,8 @@ Identity SDFFeatures::ConstructSdfLink(
   world->addRigidBody(body.get());
 
   // Generate an identity for it
-  const auto linkIdentity = this->AddLink({name, _modelID, pose, mass,
-    linkInertiaDiag, myMotionState, collisionShape, body});
+  const auto linkIdentity = this->AddLink({name, _modelID, pose, inertialPose,
```
Comment (Blast545) [inline bullet/src/SDFFeatures.cc:152]: I think you have to modify `AddLink` in the `Base.hh` file to have inertialPose as well

Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:152]: I did, it would not compile otherwise

File: bullet/src/ShapeFeatures.cc [hunk 2]
```diff
@@ -79,7 +79,9 @@ Identity ShapeFeatures::AttachMeshShape(
   const auto &modelID = linkInfo->model;
   const auto &body = linkInfo->link.get();
 
-  const auto poseIsometry = _pose;
+  auto poseWithInertia =
+    ignition::math::eigen3::convert(_pose) - linkInfo->inertialPose;
```
Comment (azeey) [inline bullet/src/ShapeFeatures.cc:83]: Same here with for `operator-`.

File: bullet/src/SDFFeatures.cc [hunk 3]
```diff
@@ -213,7 +215,8 @@ Identity SDFFeatures::ConstructSdfCollision(
     const auto &body = linkInfo->link;
     const auto &modelID = linkInfo->model;
 
-    const math::Pose3d pose = ResolveSdfPose(_collision.SemanticPose());
+    const math::Pose3d pose =
+      ResolveSdfPose(_collision.SemanticPose()) - linkInfo->inertialPose;
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:219]: You know I thought exactly the same, but tried to replicate https://github.com/osrf/gazebo/blob/67c43463c87685b7c9d485a9f8d2e6ac760815b9/gazebo/physics/bullet/BulletLink.cc#L132-L133 as much as i could

File: bullet/src/SDFFeatures.cc [hunk 4]
```diff
@@ -110,8 +110,10 @@ Identity SDFFeatures::ConstructSdfLink(
   const std::string name = _sdfLink.Name();
   const math::Pose3d pose = ResolveSdfPose(_sdfLink.SemanticPose());
   const ignition::math::Inertiald inertial = _sdfLink.Inertial();
-  auto mass = inertial.MassMatrix().Mass();
-  const auto diagonalMoments = inertial.MassMatrix().DiagonalMoments();
+  double mass = inertial.MassMatrix().Mass();
+  math::Pose3d inertialPose = inertial.Pose();
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:114]: Right, this is the step I'm missing, I need to apply the inverse in the kinematics feature. Thanks!

File: bullet/src/ShapeFeatures.cc (hunk 2 above)
Comment (Lobotuerk) [reply bullet/src/ShapeFeatures.cc]: Updated on 3ab3977

File: bullet/src/SDFFeatures.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc]: Updated on 3ab3977

File: bullet/src/KinematicsFeatures.cc [hunk 5]
```diff
@@ -52,11 +52,14 @@ FrameData3d KinematicsFeatures::FrameDataRelativeToWorld(
   const btVector3 pos = trans.getOrigin();
   const btMatrix3x3 mat = trans.getBasis();
 
-  auto eigenMat = convert(mat);
-  auto eigenVec = convert(pos);
-
-  data.pose.linear() = eigenMat;
-  data.pose.translation() = eigenVec;
+  const Eigen::Isometry3d poseIsometry =
+    ignition::math::eigen3::convert(linkInfo->inertialPose.Inverse());
```
Comment (Lobotuerk) [inline bullet/src/KinematicsFeatures.cc:56]: @azeey Is this correct? Simulation does not look like it should when using this changes

File: bullet/src/SDFFeatures.cc (hunk 4 above)
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:114]: Working on 26ab0b0

**Review by Blast545 — APPROVED** [review]
LGTM

**Pull request merged**